backend/users/services.py

```diff
@@ -1,6 +1,73 @@
+from typing import OrderedDict
+
+from presentation.models import Presentation, Tag
+from users.models import Profile, User
+
+
 class UserService:
-    def update_user_profile(self, profile, validated_data):
+    def update_user_profile(
+        self, profile: Profile, validated_data: OrderedDict
+    ) -> Profile:
         profile.file = validated_data.get("file", profile.file)
         profile.description = validated_data.get("description", profile.description)
         profile.save()
         return profile
+
+    def add_to_favourite_tags(self, user: User, validated_data: OrderedDict) -> User:
+        new_tags = validated_data.get("favourite_tags")
+        added_tags = [
```

Comment:
```suggestion
        added_tags = list(user.favourite_tags.values_list("id", flat=True))
```
should work as well here and in couple other analogue places